### DISCUSSION TITLE
how to build other language project by cake？

### DISCUSSION THREAD

**Discussion by snikeguo** [cake-build/cake#4318] [Ideas]
1. Compile cpp project with gcc?
2. Implement functions similar to kconfig and menucofig.
I want to use cake to compile Linux kernel and give up make, kconfig and menuconfig

**Comment by pascalberger** [comment]
Cake is a build orchestrator and as that agnostic to the tools it is running. While Cake itself is a .NET tool and requires .NET runtime, it can run any kind of tasks. By default Cake ships with a set of tool runners for common tools, more are available through [Extensions](https://cakebuild.net/extensions/). 

There is also the general purpose [Command alias](https://cakebuild.net/dsl/command/) and [Process alias](https://cakebuild.net/dsl/process/).

See [Reference](https://cakebuild.net/dsl/) for a full set of available commands.
